.github/workflows/catnapw.yml

```diff
@@ -18,204 +18,97 @@ on:
-  # Build Completed
-  #=====================================================================================================================
-
-  # Node 0
-  build:
-    name: Build Completed
-    needs: [build-demikernel0, build-demikernel1]
-    runs-on: ubuntu-latest
-    steps:
-      - name: Log
-        run: echo "Build Completed!"
-
-  #=====================================================================================================================
-  # Cleanup
-  #=====================================================================================================================
+jobs:
 
-  # Node 0
-  cleanup-demikernel0:
-    name: Cleanup Node 0
-    if: always()
-    needs: [build]
+  debug-pipeline:
+    name: Debug Pipeline
     runs-on: ubuntu-latest
     steps:
-    - name: Cleanup Source Tree
-      uses: appleboy/ssh-action@master
+    - name: Checkout
+      uses: actions/checkout@v3
+    - name: Setup SSH
+      shell: bash
+      run: |
+        mkdir -p $HOME/.ssh/
+        echo "${{ secrets.SSHKEY }}" > "$HOME/.ssh/id_rsa"
+        chmod 400 $HOME/.ssh/id_rsa
+        echo "Host *" > $HOME/.ssh/config
+        echo -e "\tStrictHostKeyChecking no" >> $HOME/.ssh/config
+        echo -e "\tIdentityFile $HOME/.ssh/id_rsa" >> $HOME/.ssh/config
+        echo -e "\tIdentitiesOnly yes" >> $HOME/.ssh/config
+        echo -e "\tPasswordAuthentication no" >> $HOME/.ssh/config
+        echo -e "\tUser ${{ secrets.USERNAME }}" >> $HOME/.ssh/config
+        echo -e "\tPort ${{ secrets.PORTNUM }}" >> $HOME/.ssh/config
+    - name: Run
+      run: |
+        pip install --pre azure-data-tables
+        python3 tools/demikernel_ci.py \
+          --server $SERVER \
+          --client $CLIENT \
+          --repository 'c:\demikernel' \
+          --config-path 'c:\config.yaml' \
+          --branch origin/${{ github.ref_name }} \
+          --libos $LIBOS \
+          --debug \
+          --test-unit --delay 5 \
+          --server-addr $SERVER_ADDR \
+          --client-addr $CLIENT_ADDR
+    - name: Archive Logs
+      if: always()
+      uses: actions/upload-artifact@v3
       with:
-        host: ${{ secrets.CATNAPW_HOSTNAME_A }}
-        port: ${{ secrets.PORTNUM }}
-        key: ${{ secrets.SSHKEY }}
-        username: ${{ secrets.USERNAME }}
-        script: |
-          cd c:\demikernel
-          $RustPath = Join-Path $Env:HOME "\.cargo\bin"
-          $GitPath = Join-Path $Env:ProgramFiles "\Git\cmd"
-          $Env:Path += $GitPath + ';'
-          $Env:Path += $RustPath + ';'
-          nmake clean
-          git clean -fdx
-          git checkout dev
-          git remote prune origin
-
-  # Node 1
-  cleanup-demikernel1:
-    name: Cleanup Node 1
-    if: always()
-    needs: [build]
+        name: debug-pipeline-logs
+        path: |
+          **/*.stdout.txt
+          **/*.stderr.txt
+
+  release-pipeline:
+    name: Release Pipeline
+    needs: debug-pipeline
     runs-on: ubuntu-latest
```

Comment:
Is this valid for Windows workflows?

Reply:
Yes it is. This instantiates a container on GitHub that is used as a jump box to our windows VM.

Reply:
Ok, great!